mongo/mongo.go

```diff
@@ -210,20 +210,43 @@ func transformAggregatePipeline(registry *bsoncodec.Registry, pipeline interface
 			return nil, false, fmt.Errorf("can only transform slices and arrays into aggregation pipelines, but got %v", val.Kind())
 		}
 
-		aidx, arr := bsoncore.AppendArrayStart(nil)
 		var hasOutputStage bool
 		valLen := val.Len()
 
+		switch t := pipeline.(type) {
 		// Explicitly forbid non-empty pipelines that are semantically single documents
 		// and are implemented as slices.
-		switch t := pipeline.(type) {
 		case bson.D, bson.Raw, bsoncore.Document:
 			if valLen > 0 {
 				return nil, false,
-					fmt.Errorf("%T is not an allowed pipeline type as it represents a single document. Use bson.A or mongo.Pipeline instead", t)
+					fmt.Errorf("%T is not an allowed pipeline type as it represents a single document. Use bson.A, mongo.Pipeline or bsoncore.Array instead", t)
```

Comment:
I'd leave `bsoncore.Array` out of this error message. It's an unstable/experimental type and we shouldn't really recommend it for general purpose use. Anyone who does use it probably understands BSON well enough already.

Reply:
Sounds good. Done.